docs/develop/process.md

```diff
@@ -82,7 +82,7 @@ Before starting development, make sure that you've collected feedback, that ther
 
```

Comment:
Just above:

"The goal is to avoid large changes late on in the development." -> "The goal is to avoid large changes **later** on in the development."

Given https://github.com/inveniosoftware/docs-invenio-rdm/pull/315, do you want to take a pass at the whole text and send the changes in 1 PR?